R/cell_suppression.R

```diff
@@ -252,103 +255,114 @@ suppress_secondary <- function(
 
   # Initial suppression requirement
   candidate_suppression <- suppress
-
-  for (attack.ik in ik) { # Loop over cells to be suppressed
-    for (i in seq_len(max_iter)) {
-      if (SPL[attack.ik] > 0L || UPL[attack.ik] > 0L) {
-        attacker.max <- ROI_solve(UPL_problem(candidate_suppression, attack.ik), solver = solver, ...)
-      }
-
-      if (SPL[attack.ik] > 0L || LPL[attack.ik] > 0L) {
-        attacker.min <- ROI_solve(LPL_problem(candidate_suppression, attack.ik), solver = solver, ...)
-      }
-
-      # Is the attacker subproblem successful with these constraints?
-      attacker_success <- FALSE
-
-      if (UPL[attack.ik] > 0L &&
-          identical(attacker.max$status$code, 0L) &&
-          attacker.max$objval < N[attack.ik] + UPL[attack.ik]) {
-        # Successful solution
-        alpha <- attacker.max$solution[seq_len(n)]
-        beta <- attacker.max$solution[n + seq_len(n)]
-
-        constraints(master_lp) <- c(
-          constraints(master_lp),
-          # ref, eqn (25)
-          L_constraint(
-            pmin(alpha * UB + beta * LB, UPL[attack.ik]), # Sec 4.2.1
-            dir = '>=',
-            rhs = UPL[attack.ik]
+  i <- 1L
+
+  repeat {
+
+    attacker_success <- FALSE
+
+    for (attack.ik in ik) { # Attack all cells to be suppressed
+
+      # Calculate known bounds
+        if (SPL[attack.ik] > 0L || UPL[attack.ik] > 0L) {
```

Comment:
I think something funny happened with the indentation here - could you please check the body of this for-loop is correctly indented? RStudio’s Ctrl-i should be sufficient.

Reply:
Done! I think when deleting some of the lines, I didn't re-ident with Cntl-I